### PR TITLE
.travis.yml: Upload coverage

### PR DESCRIPTION
**Pull request by abhsag24**

### CODE REVIEW

**Comment by nkprince007** [comment]
Do we just add a method to test for both try and except?

**Comment by abhsag24** [comment]
@nkprince you need to trigger the except block

File: .travis.yml [hunk 1]
```diff
@@ -7,6 +7,8 @@ sudo: true
 
 before_script:
   - pip install -r test-requirements.txt
+  - sudo apt-get install -y python3-gi python3-gi-cairo libglib2.0-dev libffi-dev libgirepository1.0 libcairo2-dev python3-cairo-dev
```
Comment (abhsag24) [inline .travis.yml]: get rid of this only added to debug

File: linux-requirements.txt [hunk 2]
```diff
@@ -1 +1 @@
-https://download.gnome.org/sources/pygobject/3.22/pygobject-3.22.0.tar.xz
\ No newline at end of file
+https://download.gnome.org/sources/pygobject/3.22/pygobject-3.22.0.tar.xz
```
Comment (abhsag24) [inline linux-requirements.txt]: remve this :P

**Comment by nkprince007** [comment]
@abhsag24, I think we need to amend our commits first
To clearly have a body specifically mentioning the changes made.
What do you think?

File: uiplib/UIP.py [hunk 3]
```diff
@@ -9,7 +9,7 @@
 
 def main():
     settingsParser = ParseSettings()
-    settings = settingsParser.settings
+    settings = settingsParser.get_settings(sys.argv[1:])
```
Comment (abhsag24) [inline uiplib/UIP.py]: why?

Comment (nkprince007) [inline uiplib/UIP.py]: To write tests for `settings.py`, the change had to be made, and this also enables a nice vision of how arguments are being handled.

Comment (abhsag24) [inline uiplib/UIP.py]: hey so i'm not sure we would want this change as stated in your other PR.

Comment (nkprince007) [inline uiplib/UIP.py]: Yeah, I'm making a few changes to faciliate `__init__` and correspond with the tests too.

File: linux-requirements.txt [hunk 4]
```diff
@@ -0,0 +1 @@
...
```
Comment (nemaniarjun) [inline linux-requirements.txt]: Would this work?
How is this supposed to be installed?
via pip or apt?

File: tests/test_scrape.py [hunk 5]
```diff
@@ -7,19 +7,40 @@
...
         self.assertEqual(scrape.get_image_links(
                             'www.reddit.com/r/CoolSite', 2),
                          [(7, 7), (8, 8)])
         self.assertEqual(scrape.get_image_links(
                             'www.unsplash.com/new', 2),
                          [(1, 1), (2, 2)])
+        self.assertEqual(scrape.get_image_links(
+                            'api.desktoppr.co/1/wallpapers', 2),
+                         [(3, 3), (4, 4)])
+
```
Comment (nemaniarjun) [inline tests/test_scrape.py]: unneeded new line?

File: linux-requirements.txt (hunk 4 above)
Comment (nkprince007) [inline linux-requirements.txt]: We actually dumped this. :/ @abhsag24 said it wasn't needed

File: tests/test_scrape.py (hunk 5 above)
Comment (nkprince007) [inline tests/test_scrape.py]: Newline is left at the end of all files, by default, as GitHub suggests

**Comment by nkprince007** [comment]
@abhsag24 is `linux-requirements.txt` still a relavent one?

**Comment by nkprince007** [comment]
ack 4f56795

File: linux-requirements.txt (hunk 4 above)
Comment (abhsag24) [inline linux-requirements.txt]: we dont need this at all

File: tests/test_scrape.py [hunk 6]
```diff
@@ -42,6 +42,10 @@ def test_get_images(self):
                   '~text?txtsize=15&txt=image1&w=120&h=120')]
             scrape.get_images('url', directory, 1)
             self.assertEqual(os.listdir(directory), ['filename.png'])
+            try:
+                os.remove('filename.png')
```
Comment (abhsag24) [inline tests/test_scrape.py]: why?

Comment (nkprince007) [inline tests/test_scrape.py]: Unnecessary file, I guess.

File: uiplib/settings.py [hunk 7]
```diff
@@ -13,18 +13,18 @@
 class ParseSettings:
     """Argument Parser class."""
 
-    def __init__(self):
+    def __init__(self, args=None):
```
Comment (abhsag24) [inline uiplib/settings.py]: unneeded change

File: uiplib/settings.py [hunk 8]
```diff
@@ -13,18 +13,18 @@
...
+        self.settings.update(self.get_settings_from_cli(args))
 
     def get_settings_from_file(self):
         """Return the settings from file."""
         with open(settings_file_path, "r") as settings_file:
             settings = json.loads(settings_file.read())
         return settings
 
-    def get_settings_from_cli(self):
+    def get_settings_from_cli(self, args=None):
```
Comment (abhsag24) [inline uiplib/settings.py]: this is not the right way to tests infact these tests are meaningless as they dont test the functionality we want them to test. Instead they test the functionality you added just so you could test it

Comment (nkprince007) [inline uiplib/settings.py]: Oh, but then how do we test it?

File: tests/test_scrape.py (hunk 6 above)
Comment (nkprince007) [inline tests/test_scrape.py]: We're working on a temporary directory here, bro. Just removing the directory soon as it finishes.

**Comment by abhsag24** [comment]
unack 58346af

**Comment by abhsag24** [comment]
unack c9a9284

**Comment by abhsag24** [comment]
ack 4f56795

**Comment by abhsag24** [comment]
unack 4e29777

**Comment by abhsag24** [comment]
ack f13f991

**Comment by abhsag24** [comment]
ack f13f991

**Comment by abhsag24** [comment]
unack d2df676

File: uiplib/scrape/scrape.py [hunk 9]
```diff
@@ -51,8 +50,8 @@ def get_unsplash_image_links(url, no_of_images):
     a_tags = soup.select('.y5w1y .hduMF .tPMQE a')
     image_links = []
     if not a_tags:
-        print('No matching image found')
-        return
+        print ('No matching image found')
```
Comment (nkprince007) [inline uiplib/scrape/scrape.py]: unnecessary spaces ?

**Comment by nkprince007** [comment]
unack c5a85dc

File: uiplib/scrape/scrape.py (hunk 9 above)
Comment (abhsag24) [inline uiplib/scrape/scrape.py]: hmm newcomer issue? :P

Comment (nkprince007) [inline uiplib/scrape/scrape.py]: But won't that break print syntax?

**Comment by nkprince007** [comment]
ack c5a85dc

File: appveyor.yml [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+  - "%CMD_IN_ENV% pip -q wheel --use-wheel --wheel-dir C:\\wheels --find-links C:\\wheels -r gui-requirements.txt"
+  - "%CMD_IN_ENV% pip -q install --find-links=C:\\wheels -r test-requirements.txt"
+  - "%CMD_IN_ENV% pip -q install --find-links=C:\\wheels -r requirements.txt"
+  - "%CMD_IN_ENV% pip -q install --find-links=C:\\wheels -r gui-requirements.txt"
+
+build: false  # Not a C# project, build stuff at the test step instead.
+
+test_script:
+  - "%CMD_IN_ENV% python -m pytest"
+  - "%CMD_IN_ENV% python setup.py install"
```
Comment (nkprince007) [inline appveyor.yml]: mind the new lines :P

Comment (nemaniarjun) [inline appveyor.yml]: This is the old file :P

**Comment by nemaniarjun** [comment]
ack c5a85dc

**Comment by abhsag24** [comment]
ack 39c68a1, ba74d2b, ba74d2b,  4546e18,

File: appveyor.yml [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+      PYTHON_ARCH: "32"
+
+    - PYTHON: "C:\\Python35-x64"
+      PYTHON_VERSION: "3.5"
+      PYTHON_ARCH: "64"
+
+branches:
+  only:
+    - master
+    - WallObj
```
Comment (abhsag24) [inline appveyor.yml]: what is this?

Comment (abhsag24) [inline appveyor.yml]: i'll ack it for now, cause it works :P

Comment (abhsag24) [inline appveyor.yml]: but we need to know whatever we are adding

**Comment by abhsag24** [comment]
ack df1f092

**Comment by abhsag24** [comment]
what happened here?

**Comment by nkprince007** [comment]
ack eb920b9

**Comment by abhsag24** [comment]
reack eb920b9, 126900, 1269001

**Comment by abhsag24** [comment]
ack 72c1066

File: tests/test_settings.py [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+            settings = self.parseSettings.get_settings_from_cli()
+            self.assertIsNotNone(settings['error'])
+
+        # No of images and offline should not be mixed
+        args = ['UIP', '--no-of-images', '20', '--offline']
+        with patch('sys.argv', new=args):
+            settings = self.parseSettings.get_settings_from_cli()
+            self.assertIsNotNone(settings['error'])
+
+        # Test-1
```
Comment (nkprince007) [inline tests/test_settings.py]: Sure, bro.

**Comment by abhsag24** [comment]
reack eb920b9

**Comment by nkprince007** [comment]
ack 92ebca5

**Comment by nkprince007** [comment]
ack 92ebca5

**Comment by nkprince007** [comment]
reack 22bbb68

File: .codecov.yml [hunk 13]
```diff
@@ -0,0 +1,9 @@
+comment: false
+coverage:
```
Comment (nkprince007) [inline .codecov.yml]: I don't understand it either.

**Comment by abhsag24** [comment]
ack a525401, 454cf34, 22bbb68,

File: .codecov.yml (hunk 13 above)
Comment (abhsag24) [inline .codecov.yml]: i think it's my fault, edited some files with vim and vim  on windows uses dos line endings b default

**Comment by abhsag24** [comment]
ack c40ff9b cmon gitmate!

**Comment by abhsag24** [comment]
ack f63f57c respond to your master!

**Comment by abhsag24** [comment]
ack c40ff9b

**Comment by abhsag24** [comment]
ack 510d560

**Comment by abhsag24** [comment]
ready 510d560

**Comment by abhsag24** [comment]
reack 510d560 the fucks your problem gitmate?

**Comment by nkprince007** [comment]
ack f63f57c

File: tests/test_scrape.py [hunk 14]
```diff
@@ -7,6 +7,17 @@
 
 class ScrapeTest(unittest.TestCase):
 
+    def setUp(self):
+        self.json = {'data' : {'children' : [{'data' : {'preview' : {
+                          'images' : [{'source' : {'url' :
+                                'url.com/some_url.png?21'}}]}}}]}}
```
Comment (abhsag24) [inline tests/test_scrape.py]: Line is longer than allowed. (83 > 79)

*LineLengthBear, severity NORMAL, section `linelength`.*

**Comment by nkprince007** [comment]
reack de756eb please gitmate

**Pull request merged**